code/datums/components/chem_booster.dm

```diff
@@ -0,0 +1,210 @@
+	///Amount of substance that the component can store
+	var/resource_storage_max = 200
+	///Ammount of sunstance stored currently
+	var/resource_storage_current
+	///Amount required for operation
+	var/resource_drain_amount = 10
+	///Opens radial menu with settings
+	var/datum/action/chem_booster/configure/configure_action
+	///Determines whether the suit is on
+	var/boost_on = FALSE
+	///Stores the value with which effect_mult is modified
+	var/boost_amount = 0
+	///Normal boost
+	var/boost_tier1 = 1
+	///Overcharged boost
+	var/boost_tier2 = 4
+	///Item connected to the system
+	var/obj/item/connected_item
+	///When was the effect activated. Used to activate negative effects after a certain amount of use
+	var/processing_start
+	COOLDOWN_DECLARE(chemboost_activation_cooldown)
+
+/datum/component/chem_booster/Initialize()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	boost_amount = 0
+
+/datum/component/chem_booster/Destroy(force, silent)
+	QDEL_NULL(configure_action)
+	wearer = null
+	return ..()
+
+/datum/component/chem_booster/RegisterWithParent()
+	. = ..()
+	configure_action = new(parent)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, list(COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT, COMSIG_ITEM_DROPPED), .proc/dropped)
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED_TO_SLOT, .proc/equipped)
+	RegisterSignal(configure_action, COMSIG_ACTION_TRIGGER, .proc/configure)
+
+/datum/component/suit_autodoc/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(
+		COMSIG_PARENT_EXAMINE,
+		COMSIG_ITEM_EQUIPPED_NOT_IN_SLOT,
+		COMSIG_ITEM_DROPPED,
+		COMSIG_ITEM_EQUIPPED_TO_SLOT))
+	QDEL_NULL(configure_action)
+
+/datum/component/chem_booster/proc/examine(datum/source, mob/user)
+	SIGNAL_HANDLER
+	to_chat(user, "<span class='notice'>The chemical system currently holds [resource_storage_current]u of green blood. Its' enhancement level is set to [boost_amount].</span>")
+
+/datum/component/chem_booster/proc/dropped(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(!iscarbon(user))
+		return
+	if(boost_on)
+		on_off()
+	if(connected_item)
+		to_chat(wearer, "<span class='warning'>You disconnect the [connected_item].</span>")
+		DISABLE_BITFIELD(connected_item.flags_item, NODROP)
+		UnregisterSignal(connected_item, COMSIG_ITEM_ATTACK)
+		connected_item = null
+	configure_action.remove_action(wearer)
+	wearer = null
+
+/datum/component/chem_booster/proc/equipped(datum/source, mob/equipper, slot)
+	SIGNAL_HANDLER
+	if(!isliving(equipper))
+		return
+	wearer = equipper
+	configure_action.give_action(wearer)
+
+/datum/component/chem_booster/process()
+	if(resource_storage_current < resource_drain_amount)
+		to_chat(wearer, "<span class='warning'>Insufficient resources to maintain operation.</span>")
+		on_off()
+	resource_storage_current -= resource_drain_amount
+
+/**
+ *	Opens the radial menu with everything
+ */
+/datum/component/chem_booster/proc/configure(datum/source)
+	SIGNAL_HANDLER_DOES_SLEEP
```

Comment:
New signal handler does sleep are not permitted

Reply:
removed this line